### PR TITLE
keeperrl: alpha28 -> alpha34

### PR DESCRIPTION
**Pull request by onny**
###### Description of changes

update package and fix build

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
build currently fails with:
```
In file included from steam_base.cpp:1:
./steam_internal.h:5:10: fatal error: 'steamworks/public/steam/steam_api_common.h' file not found
#include "steamworks/public/steam/steam_api_common.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[1]: *** [Makefile:143: obj-opt/steam_base.o] Error 1
make: *** [Makefile:130: all] Error 2
```

File: pkgs/games/keeperrl/default.nix [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
+, fetchFromGitHub
+, requireFile
+, openal
+, curl
+, libogg
+, libvorbis
+, SDL2
+, SDL2_image
+, zlib
+, clang
```
Comment (siph) [inline pkgs/games/keeperrl/default.nix:12]:
```suggestion
, clang
, libtheora
```

File: pkgs/games/keeperrl/default.nix [hunk 2]
```diff
@@ -40,7 +49,7 @@ stdenv.mkDerivation rec {
   '';
 
   buildInputs = [
-    openal curl libogg libvorbis SDL2 SDL2_image zlib
+    openal curl libogg libvorbis SDL2 SDL2_image zlib clang
```
Comment (siph) [inline pkgs/games/keeperrl/default.nix:52]:
```suggestion
    openal curl libogg libvorbis libtheora SDL2 SDL2_image zlib clang
```

**Review by siph — COMMENTED** [review]
I don't know how to add this to the review, but I got this to build by adding `libtheora` and adding `NO_STEAMWORKS=true` as a make flag.

**Comment by onny** [comment]
Thank you for the hint, updated the PR :)

**Review by siph — CHANGES_REQUESTED** [review]
This fails to build.

It needs `NO_STEAMWORKS=true` added to `makeFlags` to disable the `steamworks` integration.

Also, the pr title should be `alpha34` not `alpha35`.

**Review by siph — APPROVED** [review]
This is working for me although I have no way testing the unfree assets.

This also fixes a broken package for #199919.

**Comment by onny** [comment]
Thank you @siph for the feedback and support!

**Pull request merged**